handle_actions.go

```diff
@@ -18,50 +19,57 @@ func (ms MocksArray) DoActions(c Context) error {
 	return nil
 }
 
+// DoActions checks if a given condition matches and performs each action in m.Actions
 func (m *Mock) DoActions(c Context) error {
 	c.Values = m.Values
 	if !c.MatchCondition(m.Expect.Condition) {
 		return nil
 	}
+
 	for _, actionDispatcher := range m.Actions {
 		actualAction := getActualAction(actionDispatcher)
-		if err := actualAction.Perform(c); err != nil {
-			logrus.WithFields(logrus.Fields{
```

Comment:
move logging into `Perform` functions instead of returning an error so that we can make `ActionSendHTTP` `Perform` async